### DISCUSSION TITLE
Upgrading to new firmware

### DISCUSSION THREAD

**Discussion by jxer** [Jyers/Marlin#1012] [General Discussion]
so I am on 1.3.2, what exactly do I need to save or remember for an upgrade, I understand the mesh is saved, but is bltouch offset saved? zstop offset saved? or does this stuff all need to be re add/ adjusted every time we do a firmware upgrade?

**Comment by smooreace** [comment]
Nozzle offset, Z offset, E-steps.. All can be input into the config file
and compile in if you do it the "Right" way. Or, you can manually change
them post-flash. The mesh is NOT saved either. The first thing you do after
flashing is resetting the EEPROM data so the old garbage doesn't corrupt
the new firmware

**Reply by gothcha** [reply]
Hi just to clarify the UBL mesh IS saved but any manual mesh is not, I have flashed the firmware multiple times and the UBL mesh is preserved, ensure that the slot number is set correctly as it will default to -1 when resetting. This is a new feature post 1.3.3 when UBL is enabled.

**Reply by Tuncay-Ayhan** [reply]
You can even export the manual mesh(if that's the case) to a gcode file with Octoprint. So you can run that afterwards.


You can use this plugin for that:

[https://plugins.octoprint.org/plugins/ublmeshedit/](https://plugins.octoprint.org/plugins/ublmeshedit/)

**Reply by cruzer619** [reply]
or you can use this plugin to backup/restore the entire eeprom : https://plugins.octoprint.org/plugins/eeprom_marlin/ .  And firmware is already setup to use eeprom chat as discussed here: https://github.com/Jyers/Marlin/discussions/420
